### PR TITLE
Require julia 0.5, remove dependency on Compat

### PR DESCRIPTION
**Pull request by dmbates**

### CODE REVIEW

File: src/glmfit.jl [hunk 1]
```diff
@@ -250,7 +250,7 @@ function fit{M<:AbstractGLM,T<:FP,V<:FPVector}(::Type{M},
 end
 
 fit{M<:AbstractGLM}(::Type{M},
-    X::@compat(Union{Matrix,SparseMatrixCSC}),
```
Comment (ararslan) [inline src/glmfit.jl]: Hah, these `@compat`s are actually left over from 0.3/0.4 changes. They could have been removed a long time ago.

**Comment by andreasnoack** [comment]
LGTM

**Comment by ararslan** [comment]
Since this is bumping to Julia 0.5, would you mind changing the stuff that's deprecated in 0.5? In particular the implicit vectorization is causing a bunch of warnings.

There also seems to be a DataArrays issue on nightly... Not sure yet what that's about.

**Comment by dmbates** [comment]
@ararslan Will do the changes on implicit vectorization.

**Comment by dmbates** [comment]
I haven't been able to check this on the dev version of Julia as my compilation is failing.  I will download a pre-compiled nightly to test on.

**Comment by dmbates** [comment]
This is okay on my local version of nightly except for the problem in `DataArrays`.  Considering the switch to `CategoricalArrays` and `NullableArrays` it may not be worthwhile tracking down a problem in `DataArrays`/

**Comment by nalimilan** [comment]
Do we really need to require 0.5? Doesn't Compat provide all the needed features?

**Comment by nalimilan** [comment]
Nevermind, I hadn't noted the discussion on the other issue regarding multithreading.

**Comment by dmbates** [comment]
The checks are green on this PR now.  Can I go ahead and merge?

**Pull request merged**